src/recensio/plone/behaviors/base.py

```diff
@@ -0,0 +1,234 @@
+from plone.app.textfield import RichText as RichTextField
+from plone.app.vocabularies.catalog import CatalogSource
+from plone.app.z3cform.widget import RelatedItemsFieldWidget
+from plone.app.z3cform.widget import RichTextFieldWidget
+from plone.app.z3cform.widget import SelectFieldWidget
+from plone.autoform import directives
+from plone.autoform.interfaces import IFormFieldProvider
+from plone.dexterity.interfaces import IDexterityContent
+from plone.namedfile import field as namedfile
+from plone.supermodel import model
+from recensio.plone import _
+from z3c.form.interfaces import IAddForm
+from z3c.form.interfaces import IEditForm
+from z3c.relationfield.schema import RelationChoice
+from z3c.relationfield.schema import RelationList
+from zope import schema
+from zope.component import adapter
+from zope.interface import provider
+
+
+# from plone.supermodel.directives import fieldset
+
+
+# TODO, maybe:
+# - Move `title` to fieldset "reviewed_text"
+# - Move `subject` to fieldset "reviewed_text"
+# - Set content type for `generatedPdf` to `application/pdf`
+# - Set a character limit for `review`: 4000 characters
+# - Set rows for `review` to 20
+
+# TODO: REVIEW the following.
+# Left out attributes from recensio.contenttype.content.schemata.BaseReviewSchema:
+# - reviewAuthorLastname: For migration to reviewAuthors only
+# - reviewAuthorFirstname: For migration to reviewAuthors only
+# - uri: Partner URL is no longer used for reviews but was being kept to
+#        avoid breakage
+#        this is still used in presentations. #3103
+
+
+@provider(IFormFieldProvider)
+class IBase(model.Schema):
+    directives.widget(
+        "reviewAuthors",
+        RelatedItemsFieldWidget,
+        pattern_options={"mode": "auto", "favorites": []},
+    )
+    reviewAuthors = RelationList(
+        title=_("label_review_authors"),
+        defaultFactory=list,
+        value_type=RelationChoice(source=CatalogSource(portal_type="Person")),
+        required=True,
+    )
+
+    directives.widget("languageReview", SelectFieldWidget)
+    languageReview = schema.Choice(
+        title=_("Language(s) (review)"),
+        vocabulary="plone.app.vocabularies.SupportedContentLanguages",
+        required=False,
+        missing_value="",
+        defaultFactory=default_language,
+    )
+
+    directives.widget("languageReviewedText", SelectFieldWidget)
+    languageReviewedText = schema.Choice(
+        title=_("Language(s) (text)"),
+        vocabulary="plone.app.vocabularies.SupportedContentLanguages",
+        required=False,
+        missing_value="",
+        defaultFactory=default_language,
+    )
+
+    recensioID = schema.TextLine(
+        required=False,
+    )
+    directives.omitted("recensioID")
+
+    generatedPdf = namedfile.NamedBlobFile(
+        title=_("Generated Pdf"),
+        required=False,
+    )
+    directives.omitted("generatedPdf")
+    directives.no_omit(IEditForm, "generatedPdf")
+    directives.no_omit(IAddForm, "generatedPdf")
```

Comment:
In the old code this field is completely hidden by setting the widget condition to `False`.

https://github.com/syslabcom/recensio.contenttypes/blob/master/recensio/contenttypes/content/schemata.py#L143